dbos/system_database.py

```diff
@@ -1010,3 +1017,48 @@ def _is_buffers_empty(self) -> bool:
             len(self._workflow_status_buffer) == 0
             and len(self._workflow_inputs_buffer) == 0
         )
+
+    def enqueue(self, workflow_id: str, queue_name: str) -> None:
+        with self.engine.begin() as c:
+            c.execute(
+                pg.insert(SystemSchema.job_queue)
+                .values(
+                    workflow_uuid=workflow_id,
+                    queue_name=queue_name,
+                )
+                .on_conflict_do_nothing()
+            )
+
+    def dequeue(self, queue_name: str, concurrency: Optional[int]) -> List[str]:
+        with self.engine.begin() as c:
+            query = sa.select(SystemSchema.job_queue.c.workflow_uuid).where(
+                SystemSchema.job_queue.c.queue_name == queue_name
+            )
+            if concurrency is not None:
+                query = query.order_by(
+                    SystemSchema.job_queue.c.created_at_epoch_ms.asc()
+                ).limit(concurrency)
```

Comment:
For in-order processing, we need to make sure the `created_at_epoch_ms` order is the same as the Kafka message order. Maybe just write a tutorial on how to config Kafka broker to guarantee in-order.
We can add different policies later.

Reply:
The implementation is designed to guarantee they're the same, even in a distributed setting. We poll for the oldest uncommitted offset, enqueue it, then commit and poll again.

Reply:
I don't think you have a guarantee of in-order processing.